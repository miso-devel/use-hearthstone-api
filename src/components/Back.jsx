import { useState, useEffect } from "react";
import axios from "axios";
import { JsonBack } from "../data/back";
const APIKEY = process.env.React_APP_CARD_API_KEY;
export const Back = () => {
  const [backs, setBacks] = useState([]);
  useEffect(() => {
    const isActiveAPI = false;
    if (isActiveAPI) {
      axios
        .get(
          `https://us.api.blizzard.com/hearthstone/cardbacks/155-pizza-stone?locale=ja_JP&access_token=${APIKEY}`
        )
        .then((res) => {
          setBacks(res.data);
        });
    } else {
      setBacks(JsonBack);
    }
  }, []);

  return (
    <div className="m-10">
      <h1 className="text-center font-bold">CARD BACK</h1>
      <div className="res_card grid grid-cols-4">
        <div className="grid grid-cols-1 p-5">
          <p className="py-5 text-center">{backs.name}</p>
          <p className="py-3">{backs.text}</p>
          <img src={backs.image} alt="" />
        </div>
      </div>
    </div>
  );
};
