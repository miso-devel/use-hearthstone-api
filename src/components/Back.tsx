import React, { useState, useEffect } from "react";
import axios from "axios";
import { JsonBack } from "../data/back";

const APIKEY = process.env.React_APP_CARD_API_KEY;
export const Back = () => {
  interface JsonBacks {
    id: number;
    text: string;
    name: string;
    image: any;
  }

  const [backs, setBacks] = useState<JsonBacks>();

  useEffect(() => {
    const isActiveAPI: boolean = false;
    if (isActiveAPI) {
      axios
        .get(
          `https://us.api.blizzard.com/hearthstone/cardbacks/155-pizza-stone?locale=ja_JP&access_token=${APIKEY}`
        )
        .then((res) => {
          setBacks(res.data);
        });
    } else {
      setBacks(() => JsonBack);
    }
  }, []);

  console.log(backs);
  console.log(JsonBack);

  return (
    <div className="m-10">
      <h1 className="text-center font-bold">CARD BACK</h1>
      <div className="res_card grid grid-cols-4">
        <div className="grid grid-cols-1 p-5">
          <p className="py-5 text-center">{backs.name}</p>
          <p className="py-3">{backs.text}</p>
          <img src={backs.image} alt="" />
        </div>
      </div>
    </div>
  );
};
