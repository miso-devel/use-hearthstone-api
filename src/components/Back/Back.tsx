import React, { useState, useEffect } from "react";
import axios from "axios";
import { JsonBack } from "../../data/back";

const APIKEY = process.env.React_APP_CARD_API_KEY;
interface JsonBackType {
  id: number;
  text: string;
  name: string;
  image: any;
}
const initial = {
  id: 0,
  text: "initial text",
  name: "initial name",
  image: "a",
};
export const Back = () => {
  const [backs, setBacks] = useState<JsonBackType>(initial);
  useEffect(() => {
    const isActiveAPI: boolean = false;
    if (isActiveAPI) {
      axios
        .get(
          `https://us.api.blizzard.com/hearthstone/cardbacks/153-pizza-stone?locale=ja_JP&access_token=${APIKEY}`
        )
        .then((res: any) => {
          setBacks(res.data);
        });
    } else {
      setBacks(JsonBack);
    }
  }, []);

  return (
    <div className="">
      <h1 className="text-center font-bold">CARD BACK</h1>
      <div className="res_card grid grid-cols-4">
        <div className="grid grid-cols-1 p-5">
          <p className="py-5 text-center">{backs.name}</p>
          <p className="py-3">{backs.text}</p>
          <img src={backs.image} alt="" />
        </div>
      </div>
    </div>
  );
};
