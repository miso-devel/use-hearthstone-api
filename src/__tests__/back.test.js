import TestRenderer from "react-test-renderer";
import { Back } from "../components/Back/Back";
describe("back component", () => {
  it("render", () => {
    const tree = TestRenderer.create(<Back />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
