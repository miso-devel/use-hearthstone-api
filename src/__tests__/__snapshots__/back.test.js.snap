// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`back component render 1`] = `
<div
  className=""
>
  <h1
    className="text-center font-bold"
  >
    CARD BACK
  </h1>
  <div
    className="res_card grid grid-cols-4"
  >
    <div
      className="grid grid-cols-1 p-5"
    >
      <p
        className="py-5 text-center"
      >
        initial name
      </p>
      <p
        className="py-3"
      >
        initial text
      </p>
      <img
        alt=""
        src="a"
      />
    </div>
  </div>
</div>
`;
